### PR TITLE
Userguide: added description for MS Excel comment editing with shift+f2, explanation for carret movement when copying with nvda+f10 from start marker and aligned license description with license document

### LINKED ISSUE #6315: License documentation discrepancy
There is a discrepancy between the license as documented in Section 1.5 of the User Guide which simply links to the GPL 2.0, and going to Help > License which opens the copying.txt file and contains exceptions and other language.

### LINKED ISSUE #7394: Update User Guide to reflect change in caret location when selecting text, as per #4636
When the review cursor behaviour was changed to require pressing NVDA+f10 twice to copy (issue #4636), the caret is now moved when the text is selected.  In the explanation for the Review Cursor (second paragraph): https://www.nvaccess.org/files/nvda/documentation/userGuide.html#ReviewingText it is noted that when moving the review cursor, the system caret does not follow, so it is worth clarifying the second paragraph, possibly by adding the text:

"If selecting text that the system caret can reach, the caret moves to the selection once NVDA+f10 is pressed." after the existing paragraph so that it reads: "When moving the review cursor, the System caret does not follow along, so you can review text without losing your editing position. However, by default, when the System caret moves, the review cursor follows along. This can be toggled on and off.  If selecting text that the system caret can reach, the caret moves to the selection once NVDA+f10 is pressed".

### LINKED ISSUE #7888: User Guide: Excel's shift+f2 to enter comment editing mode.
Could you please consider to add this command(shift+f2) to User Guide 10.2. Microsoft Excel.

What's New 2014.3:
> Microsoft Excel: Provide an NVDA-specific dialog to edit the comments on the currently selected cell when pressing Excel's shift+f2 command to enter comment editing mode.

### PR DESCRIPTION
**Pull request by Adriani90**
### Link to issue number:
fixes #7888
fixes #7394
fixes #6315

### Summary of the issue:
There are some descriptions missing in the user guide. specifically:
* In MS Excel a dialog for editing a comment can be displayed when pressing shift+f2. This dialog is NVDA specific and is not implemented by MS Excel.
* When copying from start marker up to and including the review cursor with NVDA+f10, the system carret is also moved to the end of the selection in case the text is reachable.
* The description of the exceptions in the license document section.

### Description of how this pull request fixes the issue:
Added the descriptions to the coresponding sections in the user guide.

### Testing performed:
* Tested that headings show up correctly and that the key stroke shift+f2 is shown correctly.
* Tested also the shift+f2 without NVDA running and noticed the different behavior (i.e. start NVDA (I) before and (II) after pressing shift+f2 in MS Excel).
* Tested also copying from start marker with NVDA+f10 in an edit field where text is reachable by the system carret and observed that the carret was moved.

### Known issues with pull request:
None

### Change log entry:
None

### CODE REVIEW

**Comment by feerrenrut** [comment]
> shift+F2

This allows you to edit/create a comment without NVDA running also. Is this really NVDA specific? Does it really need to be mentioned in the manual since it is intended to match the behavior of Excel without NVDA? @Qchristensen your thoughts here would be appreciated.

File: user_docs/en/userGuide.t2t
Comment (Qchristensen) [inline user_docs/en/userGuide.t2t]: Everything reads well.  With shift+F2, is it worth clarifying that while NVDA overrides the dialog, shift+f2 is an Excel command (EG others differences are that shift+f2 isn't explained as an NVDA command in Input help in Excel, and you can't reassign it in Input Gestures).  Perhaps "NVDA overrides Excel's dialog for editing comments.... To add or edit a comment, in a focused cell, press shift+f2"?

**Review by Qchristensen — COMMENTED** [review]
All looks good, thanks for updating and picking up a few niggly little issues there!  I just made a small comment on the shift+f2 part.

**Comment by CyrilleB79** [comment]
Good to mention Excel's shift+F2 in User Guide.

The Shift+F2 and commenting mechanism in Excel has various aspects that may be taken into consideration:
* NVDA's shift+F2 opens an NVDA dialog that is not the same as standard Excel comment edit zone
* Excel's comment edit zone is not accessible: its content cannot be read by NVDA. I even remember that in earlier versions of Excel and/or NVDA this was making NVDA and/or Excel crash; I did not reproduce this issue anymore.
* Pressing "Add comment" in the cell's context menu is not recommended since it opens the inaccessible Excel's comment zone.
* Comments created by NVDA's dialog differ from the ones created by Excel's standard comment edit zone because standard comment zone is pre-filled with the name of the currently logged user (in bold), whereas NVDA's dialog does not contain any default text. Sighted user usually do not delete the user name information when commenting.

Feel free to add totally or partially this information in user guide if you feel it is relevant.

**Comment by Adriani90** [comment]
I would take only the part that the native comment dialog is overwritten by NVDA and the native dialog is not accessible.
I would not add any recommendations to the user guide, this is something that users should find out (i.e. through training modules from NV Access etc). If I add recommendations, then people might expect to add them also for any other software as well.
Maybe the native dialog can be made accessible at some point in time. @Qchristensen did you report this to Microsoft already?

**Comment by feerrenrut** [comment]
@Adriani90 if it's going to say that the "native dialog is not accessible" then it's only helpful to point out that the cell's context menu also opens the native dialog.

**Comment by Adriani90** [comment]
Ok I added a short note for that as well. I think this should be ok.

File: user_docs/en/userGuide.t2t
Comment (feerrenrut) [inline user_docs/en/userGuide.t2t]: I think this phrase is used in a few places, but I find it hard to read. Adding quotes around the option values would help. Overall I think it's better to just refer to the general concept and people can follow the link to get the specific details. Otherwise if the available values for this setting are changed many places in the user guide need to be updated.

File: user_docs/en/userGuide.t2t
Comment (feerrenrut) [inline user_docs/en/userGuide.t2t]: Please put each sentence on a new line for translators.

File: user_docs/en/userGuide.t2t
Comment (Adriani90) [reply user_docs/en/userGuide.t2t]: > 
> 
> I think this phrase is used in a few places, but I find it hard to read. Adding quotes around the option values would help. Overall I think it's better to just refer to the general concept and people can follow the link to get the specific details. Otherwise if the available values for this setting are changed many places in the user guide need to be updated.

Actually there are not that many places. I realized just now that there was a contradiction in the user guite. In the section "object navigation" we say that braille is tethered to "automatically" by default, whereas in the section "reviewing text" we say that by default braille is tethered to focus.

I have aligned both descriptions now, by default braille is tethered to automatically follow all the focuses and cursors.
Regarding the settings, I have removed the link strings because in my view we should focus here on readability. It is anyway not very trivial for beginners to understand how focuses and cursors work, which belongs to which system etc. I think if someone wants to jump to the settings, this can be done from the "contents" section ant the top of the page anyway.

File: user_docs/en/userGuide.t2t
Comment (feerrenrut) [inline user_docs/en/userGuide.t2t]: I think this information belongs in the #BrailleTether section. At the moment it duplicates explanations about settings and defaults. It could be simplified to something like:
```suggestion
Note [braille following object navigation is configurable. #BrailleTether]
```

Then in the #ReviewingText sections:
```suggestion
Note [braille following text review is configurable. #BrailleTether]
```

**Review by Qchristensen — APPROVED** [review]
I think that update re shift+f2 in Excel clarifies it well, great work everyone!

**Comment by dingpengyu** [comment]
hello @Adriani90 
Some checks haven’t completed yet
1 expected check
Can this problem be fixed?
thanks

**Comment by feerrenrut** [comment]
I'm still waiting on changes since my last review.

**Comment by Adriani90** [comment]
I am not having access to my local machine with the repo just now, I will do it as long as I am back home.

Thanks Reef for the suggested changes, I will accept them after syncing my forg with the current master.

**Comment by Adriani90** [comment]
Hey Reef I have committed your changes. Thanks for that. In my view the desctription already provided in #BrailleTether section is already good enough for any user to understand. Or do you think I should add there something else?

**Comment by Adriani90** [comment]
Is here anything else expected from my side? Or can this be merged as is?

**Comment by dingpengyu** [comment]
cc @feerrenrut

File: user_docs/en/userGuide.t2t
Comment (feerrenrut) [inline user_docs/en/userGuide.t2t:397]: This information belongs in the `#BrailleTether` section, as per line 366 you can replace it with something like `Note [braille following system focus and caret is configurable. #BrailleTether]`

Then make sure that the `#BrailleTether` section contains this information.

**Review by feerrenrut — CHANGES_REQUESTED** [review]

**Comment by Adriani90** [comment]
Oh I thought it would have committed your suggested changes. Apparently it it didn't commit them for whatever reason. I have addressed now the changes. Sorry for the inconvenience.

File: user_docs/en/userGuide.t2t
Comment (feerrenrut) [inline user_docs/en/userGuide.t2t:366]: Sorry to be picky again @Adriani90, but this word order: "follow automatically" is a bit strange in English. I'd remove this whole line and specify the default only once in the #BrailleTether section.
Note also, two sentences should be two lines.

```suggestion
Note: Braille following Object Navigation can be configured via [Braille Tether #BrailleTether].
```

File: user_docs/en/userGuide.t2t
Comment (feerrenrut) [inline user_docs/en/userGuide.t2t:394]: Similar problems here. Again I would just point to the braille tether section of the user guide.

```suggestion
Note: Braille following the review cursor can be configured via [Braille Tether #BrailleTether].
```

**Review by feerrenrut — CHANGES_REQUESTED** [review]

**Comment by Adriani90** [comment]
ah I forget everytime to split the sentences. Thanks for the reminder :).
I have accepted the changes and it seems they were committed now finally.

**Review by feerrenrut — APPROVED** [review]
Thanks @Adriani90

**Comment by feerrenrut** [comment]
I'm not sure why PR builds aren't being completed for this PR, my only guess is that it is because it is based on the master branch from @Adriani90 perhaps?

I have [pushed a try-build](https://ci.appveyor.com/project/NVAccess/nvda/builds/32504699)

**Comment by Adriani90** [comment]
Hey Reef, I think something went wrong in the dif history due to force pushing. I tried to rebase my branch and now I will incorporate your last suggested changes. But I am waiting first to see if checks get completed. So please don't merge yet.

**Comment by feerrenrut** [comment]
When you "accept changes" during a pull request, that creates commits only on your remote branch. If you then force push our local changes, the PR changes are lost. This is what has happened now, you will have to make the those changes again manually or reset / checkout commit 44e06f0. In the future, make sure you `git fetch` and `git merge` or `git pull` (which does both steps for you) to get remote changes onto your local branch before doing a rebase. 

I haven't checked all but it seems like at least the commits bb34160 and 44e06f0 have been lost.

**Comment by Adriani90** [comment]
Thanks for this. Yes I noticed it as well, I did fetch and rebase after that but I didn't merge before rebasing. I thought it would work that way. Anyway, I have checked and indeed only the two commits are missing from the dif, I can apply them but still the appveyor checks do not get completed. My branch is now 6 commits ahead of your master, which are exactly these 6 commits on this PR. Hmm, no idea what I should do. Maybe Close this PR and I can delete my forg and try again from scratch. Or is there another way to reset everything faster than creating everything from scratch?

**Comment by Adriani90** [comment]
Closing in favor of #11067.

**Pull request closed without merging**
